### DISCUSSION TITLE
Creation of Bruno collection

### DISCUSSION THREAD

**Discussion by danielkocot** [apideck-libraries/portman#574] [Ideas]
Bruno as an alternative is become more and more prominent in the field of API client. So what about an automated creation of a Bruno Collection by side the creation of a Postman Collection?

**Comment by thim81** [comment]
hi @danielkocot 

I have been watching Bruno as a very interesting API client and potential more open model than Postman and it's current direction.

I did some initial experimenting with Bruno in the beginning of the year. It works pretty similar like Postman and I really like the BRU language, which is very readable. 

When testing it, it felt that is was not yet on the same level as the core Postman features (with environment, testing, ...)
Especially the conversion of the Postman tests and the variable management, felt very much WIP.
Which was also confirmed by this blog article: https://danaepp.com/is-bruno-good-for-api-hacking#:~:text=Importing%20a%20Postman%20collection%20into%20Bruno

I'm planning to retry Bruno after summer, since there has been plenty of development, bug fixing and contributions happening.

When it comes to support Bruno via Portman, I have 2 options in mind:
1. Keep Portman, generating a Postman collection and use the [Bruno Importers](https://github.com/usebruno/bruno/blob/main/packages/bruno-app/src/utils/importers/postman-collection.js) to convert the generated Postman collection into a Bruno collection. But this require to have the Bruno importer better handle the Postman tests. The conversion to a Bruno collection would be build-in provided via Portman.
2. Extend Portman to natively generate the Bruno BRU files. This would mean a larger rework of the Portman internals, since we currently convert OpenAPI into Postman and use both definitions to enrich the Postman collection. To facilitate a new export format like Bruno BRU, it would make sense to rework Portman and provide "plugins" for export format (eg. export to Postman, export to Bruno, export Cypress, ....), but also think about plugin for test generation (eg. Contract tests, Content tests, ...) which would allow the community to add specific test generation plugins without making the Portman core bloated.

**Summary**: at the moment, I lend towards option 1 because it reduces complexity within Portman and development, with the requirement that Bruno improves their importers to better handle the Postman tests. AND the keep their "open" model and not got into the direction of closed source.

FYI: Opened an [feature request](https://github.com/usebruno/bruno/issues/2340) & provided an initial [PR](https://github.com/usebruno/bruno/pull/2341) on the Bruno repo, so let's see if the request gets accepted.

**Comment by thim81** [comment]
hi @danielkocot 

An update, we have a [PR](https://github.com/apideck-libraries/portman/pull/696) in the making for bringing Bruno support.

My PR for providing Bruno Converters has been accepted and implemented by the Bruno team, resulting in https://www.npmjs.com/package/@usebruno/converters

This allows for Portman to use the "Bruno Converters" to transform the generated Postman collection to a Bruno Collection.

We noticed that some Postman methods are not yet supported by Bruno, so we have created a [PR](https://github.com/usebruno/bruno/pull/4856) on the Bruno Converters package, to extend it.

Here is an example of the WIP
<img width="1573" alt="image" src="https://github.com/user-attachments/assets/8f2e1670-dd9d-463b-a313-c4359ab0793d" />
